### DISCUSSION TITLE
做了一些小调整，希望能为本项目提供参考

### DISCUSSION THREAD

**Discussion by vx13** [Meowcolm024/FGO-Automata#58] [Show and tell]
感谢作者的分享。借助这个项目，这次无限池轻松了很多。
脚本使用过程中，我个人做了一些调整。

参照： https://github.com/vx13/FGO-Automata/commits/develop

- 引入了 uiautomator2 （ https://github.com/openatx/uiautomator2 ），提高截图速度与点击速度（该框架在我的设备上需要额外权限才能进行拖拽操作，故保留原拖拽功能），但需要在手机上装额外的依赖包。
- 添加了大屏幕安卓设备自动适配（我个人用的是华为的 Matepad 10.8 ，分辨率 2560x1600 ）。设计原理是截图后按截图分辨率自动裁剪与缩放，将图片转换成 1920x1080 分辨率再进行匹配。需要执行操作时，按同样的参数做逆运算，算出真实屏幕上相应的坐标。
- 等待使用技能时，若截图未识别到目标图片，就点击左上角的空白区域，加快过图速度。
- 将助战改为助战列表，匹配助战时遍历列表里的图片。
- 将 select.png 的内容改为“请选择”，这样可以识别到宇宙凛的宝具变色技能选择框。

因为引入的依赖比较重，直接发 pull request 可能不太合适。而且我只在自己的设备上试过，不确定是否会破坏一些原有的功能（如助战选择）。
希望能请作者来参观一下调整的内容，如果能对这个项目有参考价值，我会非常荣幸。

**Comment by Meowcolm024** [comment]
我觉得像是缩放的功能和助战列表还是挺好的 :D 可以考虑pull进来
宇宙凛和红A换色卡的专门适配目前还没啥想法（懒+没时间）orz
